src/orders/components/OrderLineMetadataDialog/OrderLineMetadataDialog.tsx

```diff
@@ -0,0 +1,247 @@
+import { ButtonWithLoader } from "@dashboard/components/ButtonWithLoader/ButtonWithLoader";
+import { MetadataFormData } from "@dashboard/components/Metadata";
+import { MetadataCard } from "@dashboard/components/Metadata/MetadataCard";
+import { MetadataLoadingCard } from "@dashboard/components/Metadata/MetadataLoadingCard";
+import { DashboardModal } from "@dashboard/components/Modal";
+import { OrderLinesMetadataQuery } from "@dashboard/graphql";
+import { buttonMessages } from "@dashboard/intl";
+import { useHasManageProductsPermission } from "@dashboard/orders/hooks/useHasManageProductsPermission";
+import { mapMetadataItemToInput } from "@dashboard/utils/maps";
+import { Box, Button, Divider, Text } from "@saleor/macaw-ui-next";
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { FormattedMessage } from "react-intl";
+
+import { OrderLineDetails } from "./OrderLineDetails/OrderLineDetails";
+import { TEST_ID_ORDER_LINE_METADATA, TEST_ID_PRODUCT_VARIANT_METADATA } from "./test-ids";
+import { useHandleOrderLineMetadataSubmit } from "./useHandleSubmit";
+import { useMetadataValues } from "./useMetadataValues";
+import { useOrderMetadataFormControls } from "./useOrderMetadataFormControls";
+import { mapFieldArrayToMetadataInput } from "./utils";
+
+export type OrderMetadataDialogData = NonNullable<OrderLinesMetadataQuery["order"]>["lines"][0];
+
+export interface OrderAndVariantMetadataFormData {
+  orderLine: MetadataFormData;
+  variant: MetadataFormData;
+}
+
+interface OrderMetadataDialogProps {
+  open: boolean;
+  onClose: () => void;
+  lineId: string;
+  orderId: string;
+}
+
+export const OrderLineMetadataDialog = ({
```

Comment:
Why don't we make it generic metadata dialog? I will look the same for any other metadata.

We can also use composition for that:

```
// generic UI
declare MetadataDialog = (props: {onClose, onSave, metadata: MetadataItem[], ...}) => ReactNode

const OrderMetadataDialog = (props: {onClose}) => {
  const { data, loading } = useMetadataValues({ orderId, lineId, open });
  const { onSubmit, lastSubmittedData, submitInProgress } = useHandleOrderLineMetadataSubmit({
    initialData: data,
    onClose,
  });
  
  // other order specific

  return <MetadataDialog metadata={data} onSave={makeMutation(...)} />

}
```

so we can compose any entity metadata almost automatically